docs/quickstart.rst

```diff
@@ -0,0 +1,553 @@
+
+In the ``process_logout`` method the provider can implement some provider-specific actions such as sending a logout notification to the provider or redirecting to a SSO logout page. The ``return_url`` from argument can be passed further if the external provider allows to specify the URL to redirect to after logging out.
+
+Notice that in our example we are using ``OAuthAuthProvider`` which has no 	``process_logout`` method implemented. Therefore we are passing ``'true'`` for ``clear_session`` to remove ``'user_id'`` that we saved in ``session`` earlier and log out the user in this way. 
+
+If there is no provider specified in ``'_multipass_login_provider'`` the ``logout`` method redirects straight to the  ``return_url`` 
+
+.. _local_providers:
+
+Local providers
+********************
+
+Configuration
+~~~~~~~~~~~~~~~~
+
+This section shows an example of a configuration for an application using a local provider. If you wish to use both external and local providers, don't hesitate to specify both local and external providers in the same configuration and just follow our guide also on :ref:`external_providers`
+
+In this example ``'test_auth_provider'`` is a dummy local authentication provider, it's linked to the ``'test_identity_provider'`` as specified in ``MULTIPASS_PROVIDER_MAP``. Specifying ``'type'`` as ``'static'`` links those providers to our ``StaticAuthProvider`` and ``StaticIdentityProvider`` example classes (More on those classes later).
+
+In the ``identities`` setting of ``'test_auth_provider'`` we specify key-value pairs of username (*Pig*) and password (*pig123*), those are used for authentication by Multipass. In this example, the usernames are used as unique identifier for users. In ``identities`` settings of ``'test_identity_provider'`` we assign info keys dictionary to usernames. We also need to specify these keys in ``MULTIPASS_IDENTITY_INFO_KEYS``.
+
+.. code-block:: python
+
+	MULTIPASS_AUTH_PROVIDERS = {
+	    'test_auth_provider': {
+		'type': 'static',
+		'title': 'Insecure dummy auth',
+		'identities': {
+		    'Pig': 'pig123',
+		    'Bunny': 'bunny123'
+		}
+	    }
+	}
+
+	MULTIPASS_IDENTITY_PROVIDERS = {
+	    'test_identity_provider': {
+		'type': 'static',
+		'identities': {
+		    'Pig': {'email': 'guinea.pig@example.com', 'name': 'Guinea Pig', 'affiliation': 'Pig University'},
+		    'Bunny': {'email': 'bugs.bunny@example.com', 'name': 'Bugs Bunny', 'affiliation': 'Bunny Inc.'}
+		},
+		'groups': {
+		    'Admins': ['Pig'],
+		    'Everybody': ['Pig', 'Bunny'],
+		}
+	    }
+	}
+
+	MULTIPASS_PROVIDER_MAP = {
+	    'test_auth_provider': 'test_identity_provider'
+	}
+	
+	MULTIPASS_IDENTITY_INFO_KEYS = ['email', 'name', 'affiliation']
+
+We also need to specify the template with a login form for our provider:
+
+.. code-block:: python
+
+	MULTIPASS_LOGIN_FORM_TEMPLATE = 'login_form.html'
+	
+Implementing providers
+~~~~~~~~~~~~~~~~
```

Comment:
many of these underlines don't match the length of the line they are associated with